libsrc/apple2/ser/a2.ssc.s

```diff
@@ -239,12 +245,53 @@ SER_OPEN:
 .endif
         tax
 
+        ; Check that this works like an ACIA 6551 is expected to work
+
+        lda     ACIA_STATUS,x   ; Save current values in what we expect to be
+        sta     tmp1            ; the ACIA status register
+        lda     ACIA_CMD,x      ; and command register. So we can restore them
+        sta     tmp2            ; if this isn't a 6551.
+
+        and     #%11111110
+        sta     ACIA_CMD,x      ; Disable receiver/transmitter
+
+        lda     ACIA_CMD,x      ; Reload register
+        and     #%00000001
+        bne     NotAcia         ; We expect command register bit 0 to be 0
+
+        lda     tmp2
+        ora     #%00000001      ; Enable receiver/transmitter
+        sta     ACIA_CMD,x
+        sta     tmp3            ; Store it for comparison
+
+        lda     ACIA_CMD,x      ; Is command register what we wrote?
+        cmp     tmp3
+        bne     NotAcia
+
+        sta     ACIA_STATUS,x   ; Reset Acia (value written is not important)
+
+        lda     ACIA_CMD,x      ; Is the Acia disabled now?
+        and     #%00000001
+        beq     AciaOK
```

Comment:
I wrote before that we want to only modify bit 0. But now, I'm unsure about that:
* There are 6551 builds, which - of course totally stupidly - reset to interrrupts being **_enabled_**. Keeping interrupts enabled and then enabling the receiver/transmitter might easily result in (unhandled) interrupts.
* When we completely save and restore the two I/O port values in case it's not a 6551 then it's supposed to be not really a difference if we temporarily set other bits beside bit 0.

Therefore my proposal would be now something like this (of course with useful comments):
```
  ldy #%00000010
: tya
  sta ACIA_CMD,x
  cmp ACIA_CMD,x
  bne NotAcia
  iny
  cpy #%00000100
  bne :-
  sta ACIA_STATUS,x
  lda ACIA_CMD,x
  lsr
  bcc AciaOK
```

Reply:
Oh, that's right, I didn't think about that either. Ok then!

Reply:
Great :-) However, I haven't tested the my code proposal at all. Am I correct to presume that you did?

Reply:
I did of course! and it was correct. It does exactly what I've put in the comments. 
Driver loads on any II (c, e, gs) with an SSC in slot 2. 
Driver refuses to load on a IIgs with "Modem" instead of "Your card" in slot 2.
Driver refuses to load on a II with a non-6551 card in slot 2. I tested that on emulation, with the Pascal bytes check commented, and a CCS Model 7710 Asynchronous Serial Interface on the slot (it's based on an ACIA 6850).

Reply:
Thanks for the comprehensive testing :-)

Reply:
At that level it's better to test! 
To be sure it just doesn't happen to work, every test I do in emulation, I trace the execution to verify that it fails for the reason it should fail. For example with the simulated 6850 card, it fails at 

sta ACIA_CMD,x
cmp ACIA_CMD,x
bne NotAcia